Cubical/Foundations/Equiv/HalfAdjoint.agda

```diff
@@ -88,6 +89,25 @@ iso→HAEquiv {A = A} {B = B} (iso f g ε η) = f , (record { g = g ; sec = η ;
                                ; (j = i1) → ε (f a) k})
                       (cong (cong f) (sym (Hfa≡fHa (λ x → g (f x)) η a)) i j)
 
+-- Theorem 4.2.3. of HoTT book (similar to iso→HAEquiv)
```

Comment:
What is the difference to `iso→HAEquiv`? Is it just that this one is written using copatterns? The other one should definitely be written using copatterns as well

Reply:
It is not just the copatterns, this version computes in the proof of `vogt` later in the file, I don't know why `hcomp` does not compute though, since they should be similar if not equal.

Reply:
That seems strange indeed.. I don't think we should have two versions of the same result

Reply:
Btw, what do you mean with "does not compute"?

Reply:
By compute I mean that hcomp reduces to a simple term, so that I can do reflexivity or a simple rewrite.

Reply:
Ok, so the problem is that if you use the other proof then  unfolding it leads to something that doesn't let you easily prove `vogt`?

Reply:
Yes, exactly.

Reply:
Ok, let's keep both version for now and I'll take a look at it when this PR is merged. Please add a comment above `iso→HAEquiv'` explaining why it is there so that we don't forget